server/pairing/payload_manager.go

```diff
@@ -49,18 +54,71 @@ type PayloadManager interface {
 	LockPayload()
 }
 
+type PayloadSourceSenderConfig struct {
+	KeyUID   string `json:"keyUID"`
+	Password string `json:"password"`
+}
+
+type PayloadSourceReceiverConfig struct {
+	KDFIterations int `json:"kdfIterations"`
+	NodeConfig    *params.NodeConfig
+	// this field already exists within params.NodeConfig, but it doesn't support json marshalling, so we need to duplicate it here
+	RootDataDir string
+	// corresponding to field current_network from table settings, so that we can override current network from sender
+	SettingCurrentNetwork string
+}
+
```

Comment:
Rather than these 2 structs being embedded in `PayloadSourceConfig`, perhaps we should inverse the relationship.

A `PayloadSourceSenderConfig` should have a `PayloadSourceConfig` because it is a kind of `PayloadSourceConfig`
A `PayloadSourceReceiverConfig ` should have a `PayloadSourceConfig` because it is a kind of `PayloadSourceConfig`

But a `PayloadSourceConfig` is not a kind of `PayloadSourceSenderConfig` or `PayloadSourceReceiverConfig`

Reply:
Hmm, I notice that even with the changes in this PR that `PayloadSourceConfig` will only be left with one field `KeystorePath` which is used in two different ways:

- With keyUID
- Without keyUID

So perhaps the time of `PayloadSourceConfig` is over now.

Reply:
Hmmm. If / when we merge my latest PR https://github.com/status-im/status-go/pull/3192/files#diff-737158e6a8fd20bd9226198c0c4ff00ae4d5c129c455a790194de3503b02bf4b my new field makes no sense in the context of `PayloadSourceConfig`. Because this change is not related to the payload source, it is specifically related to the `PairingServer` configuration and not `Sender` / `Receiver` configuration.

```go
	// Timeout the number of milliseconds after which the pairing server will automatically terminate
	Timeout uint `json:"timeout"`
```

We should probably consider redesigning the configuration schema, because we are starting to grow it into more complex use cases.

Reply:
> Rather than these 2 structs being embedded in `PayloadSourceConfig`, perhaps we should inverse the relationship.
> 
> A `PayloadSourceSenderConfig` should have an embedded `PayloadSourceConfig` because it is a kind of `PayloadSourceConfig` A `PayloadSourceReceiverConfig ` should have an embedded `PayloadSourceConfig` because it is a kind of `PayloadSourceConfig`
> 
> But a `PayloadSourceConfig` is not a kind of `PayloadSourceSenderConfig` or `PayloadSourceReceiverConfig`, does it make sense to embed either?

FYI, I added these 2 types of config(sender/receiver) because i saw a problem that the param `configJSON` is not very clear for our relate API(GetConnectionStringForBeingBootstrapped/InputConnectionStringForBootstrapping etc.) in status.go

Reply:
> Hmmm. If / when we merge my latest PR https://github.com/status-im/status-go/pull/3192/files#diff-737158e6a8fd20bd9226198c0c4ff00ae4d5c129c455a790194de3503b02bf4b my new field makes no sense in the context of `PayloadSourceConfig`. Because this change is not related to the payload source, it is specifically related to the `PairingServer` configuration and not `Sender` / `Receiver` configuration.
> 
> ```go
> 	// Timeout the number of milliseconds after which the pairing server will automatically terminate
> 	Timeout uint `json:"timeout"`
> ```
> 
> We should probably consider redesigning the configuration schema, because we are starting to grow it into more complex use cases.

agree, we need a refactor

Reply:
Hmm . We could have `PairingConfig` which is a global level config, with sub configs. But then I'll be a hypocrite and completely contradict my whole point I'm making here https://github.com/status-im/status-go/pull/3179/files#r1107080212

Reply:
for the time being, can we leave the refactor in a separate PR? 🤔

Reply:
Yes, that's fine I'll raise an issue on this point

Reply:
thank you